doc/readthedocs/developer_dirstructure.rst

```diff
@@ -4,22 +4,65 @@ The |FOURC| code comes with documentation, example input files and
 support scripts. The important subdirectories are the following:
 
 
-:``src``: contains the real |FOURC| code in several subdirectories
+:``apps``: Contains executables built on top of the |FOURC| library code.
 
-:``tests/input_files``:   contains various valid and running ``*``\ ``.dat`` files,
-    |FOURC| input files that are used for (automatic) testing.
+:``src``: Contains the bulk of the |FOURC| code organized into several modules.
 
-:``utilities``:  contains configuration scripts needed to setup a |FOURC| Makefile,
+:``tests/input_files``:   Contains various valid and running input files.
+    These input files are also used for automated testing.
 
-:``tests``:   contains tests that are not directly related to |FOURC|, but rather to the input
+:``utilities``:  Contains configuration scripts needed to setup a |FOURC| Makefile,
+
+:``tests``:   Contains tests that are not directly related to |FOURC|, but rather to the input
     and output, balso also comprising the whole tool chain (framework tests).
 
-:``unittests``:  contains the source files for stand alone tests aka unittests.
+:``unittests``:  [Legacy] Contains the source files for stand alone tests aka unittests.
+    These tests will move closer to the source code into the respective modules.
+
+:``doc``:   Contains this documentation and also the setup for Doxygen.
+
+:``presets``:   Contains preset files for CMake.
+
+:``docker``: Contains the setup files for docker images for running |FOURC| inside of it.
+
+Details of the ``src`` directory
+--------------------------------
+
+The ``src`` directory contains the modules that make up the |FOURC| library. Each module is
+split into a ``src`` and ``tests`` directory. The ``src`` directory contains the production code
+of the module, while the ``tests`` directory contains the module-related tests.
```

Comment:
Do we want to mention the `Core` here? Since it is, per definition, a collection of modules?

Reply:
I briefly thought about it. `core` is a large module with a (too?) complicated substructure. I went with this basic description since I think we need to explore a bit more how exactly we want the structure to look in the end.